lib/shopify_app/app_proxy_verification.rb

```diff
@@ -0,0 +1,32 @@
+module ShopifyApp
+  module AppProxyVerification
+    extend ActiveSupport::Concern
+
+    # Usage:
+    #   include ShopifyApp::AppProxyVerification
+    #   before_action :verify_proxy_request
+    def verify_proxy_request
+      return head :unauthorized unless query_string_valid?(request.query_string)
+    end
```

Comment:
I'm sorry but I don't understand this comment - can you explain what you mean again?

Also our style would be to not have any comments in the code